### ISSUE TITLE
Wrong formatting of `are` in some specific cases

### ISSUE THREAD

**Issue by RolT** [kkinnear/zprint#212]
Hello, I encountered a strange formatting using zprint recently. I tried to reproduce it with a simpler code and the default config:

```clojure
(let [a '(deftest test
           (are [a b c d] (= a b c d)
             "1"    "1" "1" "1"
             "1"    "1" "ABCDEFGHI" "1"
             "1"    "1" "ABCDEFGHI" "1"))]
  (println (zp/zprint-file-str (str a) "")))

(deftest test
  (are [a b c d] (= a b c d)
   "1"   "1"   "1"
   "1"   "1"   "1"
   "ABCDEFGHI" "1" "1"
   "1" "ABCDEFGHI" "1"))
```

Each line contains 3 tokens instead of 4.  

A slightly different one gets formatted correctly:

```clojure
(let [a '(deftest test
           (are [a b c d] (= a b c d)
             "1"    "1" "1" "1"
             "1"    "1" "ABCDEFGH" "1"
             "1"    "1" "ABCDEFGH" "1"))]
  (println (zp/zprint-file-str (str a) "")))

(deftest test
  (are [a b c d] (= a b c d)
   "1"   "1"   "1"          "1"
   "1"   "1"   "ABCDEFGH"   "1"
   "1"   "1"   "ABCDEFGH"   "1"))
```

**Comment by kkinnear** [comment]
Wow, that's a good one!  Thank you so much for noticing it and creating an issue!  Totally reproducible, fortunately.  I haven't gotten to the end of it, but I have found that the problem is caused by something in the code that does the justification.  So, you can always turn off justification and still get some good formatting for `are` functions:

This is equivalent to the default `:fn-map` for `are`, which demonstrates the problem, albeit with a slightly different example than you gave:
```
zprint.core=> (czprint "(are [a b c d] (= a b c d) 1 1 1 1 1 1 1234567890 1)" {:parse-string? true :fn-map {"are" [:guided {:style :areguide}]}})
(are [a b c d] (= a b c d)
  1 1 1
  1 1 1
  1234567890 1)
```
This is an alternative `:fn-map` for `are`, which doesn't justify, and it works correctly:
```
zprint.core=> (czprint "(are [a b c d] (= a b c d) 1 1 1 1 1 1 1234567890 1)" {:parse-string? true :fn-map {"are" [:guided {:style :areguide-nj}]}})
(are [a b c d] (= a b c d)
  1 1 1 1
  1 1 1234567890 1)
```
You can just put `{:fn-map {"are" [:guided {:style :areguide-nj}]}}` in your configuration to get this behavior.

I'm looking forward to figuring out what is going wrong here, as it certainly isn't obvious!  Thanks again!

**Comment by kkinnear** [comment]
The problem here occurs when justification fails because the variance of the lengths of the elements in a column exceeds the configured `:max-variance`.   This is why the difference in length in your examples caused different behavior.  So, not justifying fixes this problem (see above).

If you like justification, you can also still get it without this issue occurring if you allow any level of variance, like this:
```
 % ./zprintm-1.2.0 '{:pair {:justify {:max-variance 1000}}}'  <cljfiles/i212.clj
(deftest test
  (are [a b c d] (= a b c d)
    "1"   "1"   "1"           "1"
    "1"   "1"   "ABCDEFGHI"   "1"
    "1"   "1"   "ABCDEFGHI"   "1"))
```
The only issue with this is that the sizes of strings are wrong, and every string is seen to be 2 characters longer than it really is.  That causes extra space between the columns, but you still get justification, which can be really nice. Sp, if you like the justification and can live with the extra spaces after strings, then you can put this:
```
{:pair {:justify {:max-variance 1000}}}
```
in your configuration and you will get it.

Both of these issues -- the mishandling of the number of rows (which you reported) and the additional space allotted to strings (which I noticed while debugging the rows issue) have been fixed and will be in the next release, which I expect to be `1.2.1`.

Thanks again for reporting this!

**Comment by RolT** [comment]
Thanks, I changed the code structure a bit to avoid this problem before submitting the issue. But I'll keep that tip in mind if I encounter it in the future.

**Comment by kkinnear** [comment]
Fixed in `1.2.1`.  Thanks again for noticing this!
